### PR TITLE
Improve Documentation

### PR DESCRIPTION
**Pull request by fitzthum**
General improvements to KBS documentation

~~WIP for now.~~

@bpradipt @tylerfanelli lmk if there is anything you'd like me to add

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,50 @@
...
+
+[Reference Value Provider Service](attestation-service/rvps)
+The RVPS manages reference values used to verify TEE evidence. 
+This is [mentioned](https://www.ietf.org/archive/id/draft-ietf-rats-architecture-22.html#name-endorser-reference-value-pr)
+in the RATS model as well.
+
+[Client Tool](kbs/tools/client)
+This is a simple tool which can be used to test or configure the KBS and AS.
+
+##
```
Comment (Xynnn007) [inline README.md:31]: Do you think we need to add a topic like `Architecture` here?

**Review by Xynnn007 — COMMENTED** [review]
Great, this will make the repo more new-comer friendly

File: README.md [hunk 2]
```diff
@@ -1,18 +1,50 @@
...
+
+These components are flexible and can be deployed in several different configurations.
+This figure shows one common way to deploy these components in conjunction with certain guest components.
+
+<div align=center>
+
+![](kbs/docs/pictures/cluster.svg)
+
+</div>
```
Comment (bpradipt) [inline README.md:40]: I'm thinking if skopeo and encrypted container image is really needed. We can just mentioned loading of keys in the right path to keep it simple

File: README.md [hunk 3]
```diff
@@ -1,18 +1,50 @@
...
+in the RATS model.
 
 [Attestation Service](attestation-service)
-Attestation Service (AS for short) is a general function set that can verify TEE evidence.
+The AS verifies TEE evidence.
+In the RATS model this is a [Verifier](https://www.ietf.org/archive/id/draft-ietf-rats-architecture-22.html#name-verifier)
+
+[Reference Value Provider Service](attestation-service/rvps)
+The RVPS manages reference values used to verify TEE evidence. 
+This is [mentioned](https://www.ietf.org/archive/id/draft-ietf-rats-architecture-22.html#name-endorser-reference-value-pr)
```
[the remark this answers is not in the document]
Comment (fitzthum) [reply README.md:25]: Ok, I'll change this to say that it is related to that section of the doc.

File: README.md (hunk 2 above)
Comment (fitzthum) [reply README.md]: Ok, I replaced the diagram with a fancy mermaid one. I am using a flowchart, which has some limitations in terms of layout. Mermaid also supports more complex C4 diagrams, but only as an experimental feature. What do you think about the flow chart? cc @Xynnn007

Comment (bpradipt) [reply README.md]: Looks good to me.

File: attestation-service/README.md [hunk 4]
```diff
@@ -134,16 +159,19 @@ Supported Verifier Drivers:
 
 The AS supports modular policy engine, which can be specified through the AS configuration. The currently supported policy engines are:
```
Comment (bpradipt) [inline attestation-service/README.md:160]: Should there be a mention that the policy is universal and no support for multiple policies?

File: attestation-service/README.md [hunk 5]
```diff
@@ -1,31 +1,60 @@
...
+2. Evaluate the claims presented by the evidence (i.e. check that measurements are what the client expects).
 
-# Components
+The first step is accomplished by one of the platform-specific [Verifier Drivers](#verifier-drivers).
+The second step is driven by the [Policy Engine](#policy-engine) with help from the [RVPS](#reference-value-provider-service).
 
+# Usage
 ## Library
 
 The AS can be built and imported as a Rust crate into any project providing attestation services.
```
Comment (bpradipt) [inline attestation-service/README.md:52]: Should there be any mention of CoCo AS and Amber/IntelTrustAuthority AS?

Comment (fitzthum) [reply attestation-service/README.md:52]: ITA is supported in the KBS as an alternative to the AS. The AS itself doesn't have any support for it. We do have some support for veraison in the AS, but it's just part of the CCA verifier.

File: attestation-service/README.md (hunk 4 above)
Comment (fitzthum) [reply attestation-service/README.md:160]: There is support for multiple policies. I am going to change this sentence around though. I don't think the modularity of the policy engine (i.e. the possibility to implement something other than OPA) is very significant.

**Comment by fitzthum** [comment]
Ok, I think this is enough for now. PTAL @Xynnn007 and @bpradipt

File: README.md [hunk 6]
```diff
@@ -1,18 +1,70 @@
...
-The Confidential Containers Key Broker Service (KBS) is a remote server which facilitates remote attestation, which is the reference implementation of [Relying Party](https://www.ietf.org/archive/id/draft-ietf-rats-architecture-22.html)
-and [Verifier](https://www.ietf.org/archive/id/draft-ietf-rats-architecture-22.html) in [RATS](https://datatracker.ietf.org/doc/draft-ietf-rats-architecture/)
-role terminology.
+## Components
 
-[Attestation Service](attestation-service)
-Attestation Service (AS for short) is a general function set that can verify TEE evidence.
+- [Key Broker Service](kbs)
+The KBS is a server that facilitates remote attestation and secret delivery.
+It's role is similar to that of the [Relying Party](https://www.ietf.org/archive/id/draft-ietf-rats-architecture-22.html#name-relying-party)
```
Comment (Xynnn007) [inline README.md:16]:
```suggestion
Its role is similar to that of the [Relying Party](https://www.ietf.org/archive/id/draft-ietf-rats-architecture-22.html#name-relying-party)
```

**Review by Xynnn007 — APPROVED** [review]
lgtm. only a nit

**Review by bpradipt — APPROVED** [review]
/lgtm

**Pull request merged**